src/components/MapView/Analyser/index.tsx

```diff
@@ -78,32 +81,128 @@ function Analyser({ classes }: AnalyserProps) {
+  );
+  const [statistic, setStatistic] = useState(
+    (urlParams[UrlParamsEnum.STATISTIC] as AggregationOperations) ||
+      AggregationOperations.Mean,
+  );
+  const [baselineLayerId, setBaselineLayerId] = useState<LayerKey>(
+    urlParams[UrlParamsEnum.BASELINE_LAYER] as LayerKey,
+  );
+  const [selectedDate, setSelectedDate] = useState<number | null>(
+    urlParams[UrlParamsEnum.DATE]
+      ? new Date(urlParams[UrlParamsEnum.DATE] as string).getTime()
+      : null,
+  );
+  const [belowThreshold, setBelowThreshold] = useState(
+    urlParams[UrlParamsEnum.BELOW_THRESHOLD] || '',
+  );
+  const [aboveThreshold, setAboveThreshold] = useState(
+    urlParams[UrlParamsEnum.ABOVE_THRESHOLD] || '',
+  );
   const [thresholdError, setThresholdError] = useState<string | null>(null);
 
+  const adminBoundariesExtent = useMemo(() => {
+    if (!boundaryLayerData) {
+      // not loaded yet. Should be loaded in MapView
+      return null;
+    }
+    return bbox(boundaryLayerData.data) as Extent; // we get extents of admin boundaries to give to the api.
+  }, [boundaryLayerData]);
+
+  // Used with a link
+  const runAnalyser = async () => {
+    if (!adminBoundariesExtent) {
+      return;
+    } // hasn't been calculated yet
+
+    if (!selectedDate) {
+      throw new Error('Date must be given to run analysis');
+    }
+
+    if (!hazardLayerId || !baselineLayerId) {
+      throw new Error('Layer should be selected to run analysis');
+    }
+
+    const selectedHazardLayer = LayerDefinitions[
+      hazardLayerId
+    ] as WMSLayerProps;
+    const selectedBaselineLayer = LayerDefinitions[
+      baselineLayerId
+    ] as NSOLayerProps;
+
+    const params: AnalysisDispatchParams = {
+      hazardLayer: selectedHazardLayer,
+      baselineLayer: selectedBaselineLayer,
+      date: selectedDate,
+      statistic,
+      extent: adminBoundariesExtent,
+      threshold: {
+        above: parseFloat(aboveThreshold) || undefined,
+        below: parseFloat(belowThreshold) || undefined,
+      },
+    };
+
+    await dispatch(requestAndStoreAnalysis(params));
+  };
+
+  const [isUrlActionPlayed, setIsUrlActionPlayed] = useState(false);
+  useEffect(() => {
+    if (
+      urlPath === '/analysis' &&
+      !isUrlActionPlayed &&
+      adminBoundariesExtent
+    ) {
+      setIsUrlActionPlayed(true);
+      runAnalyser().then(_ => setIsAnalyserFormOpen(true));
+    }
+  }, [
+    adminBoundariesExtent,
+    isUrlActionPlayed,
+    runAnalyser,
+    urlParams,
+    urlPath,
+  ]);
+
   // set default date after dates finish loading and when hazard layer changes
   useEffect(() => {
     const dates = hazardLayerId
       ? availableDates[
           (LayerDefinitions[hazardLayerId] as WMSLayerProps)?.serverLayerName
         ]
       : null;
-    if (!dates || dates.length === 0) {
-      setSelectedDate(null);
-    } else {
-      setSelectedDate(dates[dates.length - 1]);
+    if (!urlParams[UrlParamsEnum.DATE]) {
```

Comment:
This useEffect is responsible for selecting a default date on the datepicker when the user changes their hazard layer selection.
This isn't a good idea because now, when the user selects a new layer, the date picker doesn't update with compatible dates. 
Since it'll be hard to make this use effect work with URL Queries, the best way I see is to edit the  `LayerDropdown` `setValue` function and add the functionality there, so when a new value is picked, the date selector is updated as well from before.